guiEditor/src/components/commandBarComponent.tsx

```diff
@@ -13,10 +13,33 @@ interface ICommandBarComponentProps {
 }
 
 export class CommandBarComponent extends React.Component<ICommandBarComponentProps> {
-
+    private _panning: boolean;
+    private _zooming: boolean;
+    private _selecting: boolean;
     public constructor(props: ICommandBarComponentProps) {
         super(props);
 
+        props.globalState.onPanObservable.add(() => {
+            this._panning = !this._panning;

```

Comment:
have you tried passing those to the component state instead of forcing an update? the component's state should have those three elements, and when you set state the component will reload by itself.

Reply:
And every time you really want to use `forceUpdate`, do whatever you can not to (like using setState, as an example) :-)